components/atom/input/src/Mask/index.js

```diff
@@ -1,50 +1,69 @@
-import {forwardRef, useEffect, useRef, useState} from 'react'
+import {forwardRef, useEffect} from 'react'
+import {useIMask} from 'react-imask'
 
 import PropTypes from 'prop-types'
 
-import Input from '../Input/index.js'
+import useMergeRefs from '@s-ui/react-hooks/lib/useMergeRefs'
 
-const MaskInput = forwardRef(
-  ({name, onChange, mask: maskOptions, ...props}, forwardedRef) => {
-    const [mask, setMask] = useState(null)
-    const refInput = useRef(null)
-
-    useEffect(() => () => mask && mask.destroy(), [mask])
+import Input from '../Input/Component/index.js'
 
-    const handleChange = (ev, {value}) => {
-      typeof onChange === 'function' && onChange(ev, {value})
-    }
-
-    const handleFocus = () => {
-      if (!mask) {
-        import('imask').then(({default: IMask}) => {
-          setMask(new IMask(refInput.current, maskOptions))
-        })
+const MaskInput = forwardRef(
+  (
+    {
+      name,
+      onChange,
+      onComplete,
+      mask,
+      value: propValue,
+      defaultValue,
+      ...props
+    },
+    forwardedRef
+  ) => {
+    const {
+      ref: refInput,
+      value: maskedValue = '',
+      setValue
+    } = useIMask(
+      {...mask},
+      {
+        onAccept: (value, maskRef, event, ...args) => {
+          typeof onChange === 'function' &&
+            onChange(event, {value, maskRef, ...args})
+        },
+        onComplete: (value, maskRef, event, ...args) => {
+          typeof onComplete === 'function' &&
+            onComplete(event, {value, maskRef, ...args})
+        }
```

Comment:
**nit**: may improve the readability abstracting a `isFunction` helper function.